packages/dependency-extraction-webpack-plugin/lib/index.js

```diff
@@ -352,12 +352,25 @@ class DependencyExtractionWebpackPlugin {
 	/**
 	 * Can we trace a line of static dependencies from an entry to a module
 	 *
-	 * @param {webpack.Compilation}       compilation
-	 * @param {webpack.DependenciesBlock} block
+	 * @param {webpack.Compilation}            compilation
+	 * @param {webpack.DependenciesBlock}      block
+	 * @param {Set<webpack.DependenciesBlock>} visited
```

Comment:
It might be better to only store IDs of each visited block instead of the whole block. This would use less memory.  

However, I could not see if the `DependenciesBlock` has a stable ID of any kind: https://github.com/webpack/webpack/blob/899f06934391baede59da3dcd35b5ef51c675dbe/lib/DependenciesBlock.js#L29

Without benchmarking, hard to know if it makes a difference.

Reply:
I have some ideas for this, I'm working on it now.